File: src/algebra1_exponent_rules_easy.py
"""
algebra1 - exponent_rules (easy)
Generated: 2026-02-22T03:57:45.740326
"""

from problem_utils import *

BASES = [2, 3, 4, 5, 6, 7, 8, 9, 10]
VARS = [x, y, a, b]

def generate():
    problem_type = choice(['simplify_product', 'simplify_quotient', 'power_of_power',
                           'evaluate_small', 'zero_exponent', 'product_coeffs'])

    if problem_type == 'simplify_product':
        # x^a * x^b = x^(a+b) with varied bases and exponents
        base_var = choice(VARS)
        exp1 = randint(1, 8)
        exp2 = randint(1, 8)
        expr = base_var**exp1 * base_var**exp2
        ans = base_var**(exp1 + exp2)

        return problem(
            question=f"Simplify: ${latex(expr)}$",
            answer=ans,
            difficulty=(1000, 1200),
            topic="algebra1/exponent_rules",
            solution=steps(
                f"Product rule: $x^a \\cdot x^b = x^{{a+b}}$",
                f"${latex(base_var)}^{{{exp1}}} \\cdot {latex(base_var)}^{{{exp2}}} = {latex(base_var)}^{{{exp1}+{exp2}}} = {latex(ans)}$"
            )
        )

    elif problem_type == 'simplify_quotient':
        # x^a / x^b = x^(a-b) where a > b
        base_var = choice(VARS)
        exp1 = randint(3, 10)
        exp2 = randint(1, exp1-1)
        ans = base_var**(exp1 - exp2)

        return problem(
            question=f"Simplify: $\\frac{{{latex(base_var**exp1)}}}{{{latex(base_var**exp2)}}}$",
            answer=ans,
            difficulty=(1100, 1250),
            topic="algebra1/exponent_rules",
            solution=steps(
                f"Quotient rule: $\\frac{{x^a}}{{x^b}} = x^{{a-b}}$",
                f"$\\frac{{{latex(base_var)}^{{{exp1}}}}}{{{latex(base_var)}^{{{exp2}}}}} = {latex(base_var)}^{{{exp1}-{exp2}}} = {latex(ans)}$"
            )
        )

    elif problem_type == 'power_of_power':
        # (x^a)^b = x^(a*b)
        base_var = choice(VARS)
        exp1 = randint(2, 6)
        exp2 = randint(2, 5)
        ans = base_var**(exp1 * exp2)

        return problem(
            question=f"Simplify: $({latex(base_var)}^{{{exp1}}})^{{{exp2}}}$",
            answer=ans,
            difficulty=(1050, 1200),
            topic="algebra1/exponent_rules",
            solution=steps(
                f"Power rule: $(x^a)^b = x^{{ab}}$",
                f"$({latex(base_var)}^{{{exp1}}})^{{{exp2}}} = {latex(base_var)}^{{{exp1} \\cdot {exp2}}} = {latex(ans)}$"
            )
        )

    elif problem_type == 'evaluate_small':
        # Evaluate numeric expression: base^exp
        base = choice(BASES)
        exp = randint(1, 4)
        ans = base**exp

        return problem(
            question=f"Evaluate: ${base}^{{{exp}}}$",
            answer=ans,
            difficulty=(1000, 1100),
            topic="algebra1/exponent_rules",
            solution=steps(
                f"${base}^{{{exp}}}$ means multiply ${base}$ by itself ${exp}$ time(s)",
                f"$= {ans}$"
            )
        )

    elif problem_type == 'zero_exponent':
        # x^0 = 1 for any nonzero x
        base_var = choice(VARS)
        coeff = randint(2, 10)
        # coeff * x^0 = coeff
        ans = coeff

        return problem(
            question=f"Simplify: ${coeff} \\cdot {latex(base_var)}^{{0}}$",
            answer=ans,
            difficulty=(1000, 1100),
            topic="algebra1/exponent_rules",
            solution=steps(
                f"Any nonzero expression raised to the power $0$ equals $1$: ${latex(base_var)}^{{0}} = 1$",
                f"${coeff} \\cdot {latex(base_var)}^{{0}} = {coeff} \\cdot 1 = {coeff}$"
            )
        )

    else:  # product_coeffs
        # c1*x^a * c2*x^b = (c1*c2)*x^(a+b)
        base_var = choice(VARS)
        c1 = randint(2, 6)
        c2 = randint(2, 6)
        exp1 = randint(1, 6)
        exp2 = randint(1, 6)
        ans = (c1*c2) * base_var**(exp1+exp2)

        return problem(
            question=f"Simplify: $({c1}{latex(base_var)}^{{{exp1}}})({c2}{latex(base_var)}^{{{exp2}}})$",
            answer=ans,
            difficulty=(1100, 1250),
            topic="algebra1/exponent_rules",
            solution=steps(
                f"Multiply coefficients and apply product rule for exponents",
                f"$= ({c1} \\cdot {c2}) \\cdot {latex(base_var)}^{{{exp1}+{exp2}}}$",
                f"$= {c1*c2}{latex(base_var)}^{{{exp1+exp2}}}$"
            )
        )

emit(generate())

File: src/algebra1_exponent_rules_easy.jl
# algebra1 - exponent_rules (easy)
# Generated: 2026-03-08T19:58:22.178011

include(joinpath(@__DIR__, "..", "julia", "src", "ProblemUtils.jl"))
using .ProblemUtils

@script x begin
    set_topic!("algebra1/exponent_rules")
    
    problem_type = rand(1:6)
    
    if problem_type == 1
        # Product rule: x^a * x^b = x^(a+b)
        a = randint(2, 8)
        b = randint(2, 8)
        ans = a + b
        
        problem(
            question="Simplify: \$x^{$a} \\cdot x^{$b}\$",
            answer="x^$ans",
            difficulty=(700, 900),
            solution=steps(
                "Use the product rule: \$x^a \\cdot x^b = x^{a+b}\$",
                sol("Combine exponents", "x^{$a + $b}"),
                sol("Answer", "x^{$ans}")
            ),
            time=45
        )
        
    elseif problem_type == 2
        # Quotient rule: x^a / x^b = x^(a-b)
        a = randint(5, 12)
        b = randint(2, a-1)
        ans = a - b
        
        problem(
            question="Simplify: \$\\frac{x^{$a}}{x^{$b}}\$",
            answer="x^$ans",
            difficulty=(750, 950),
            solution=steps(
                "Use the quotient rule: \$\\frac{x^a}{x^b} = x^{a-b}\$",
                sol("Subtract exponents", "x^{$a - $b}"),
                sol("Answer", "x^{$ans}")
            ),
            time=45
        )
        
    elseif problem_type == 3
        # Power rule: (x^a)^b = x^(ab)
        a = randint(2, 6)
        b = randint(2, 5)
        ans = a * b
        
        problem(
            question="Simplify: \$(x^{$a})^{$b}\$",
            answer="x^$ans",
            difficulty=(750, 950),
            solution=steps(
                "Use the power rule: \$(x^a)^b = x^{ab}\$",
                sol("Multiply exponents", "x^{$a \\cdot $b}"),
                sol("Answer", "x^{$ans}")
            ),
            time=45
        )
        
    elseif problem_type == 4
        # Combined: (x^a * x^b)^c
        a = randint(1, 4)
        b = randint(1, 4)
        c = randint(2, 4)
        ans = (a + b) * c
        
        problem(
            question="Simplify: \$(x^{$a} \\cdot x^{$b})^{$c}\$",
            answer="x^$ans",
            difficulty=(900, 1100),
            solution=steps(
                "First apply product rule inside: \$x^{$a} \\cdot x^{$b} = x^{$(a+b)}\$",
                "Then apply power rule: \$(x^{$(a+b)})^{$c} = x^{$(a+b) \\cdot $c}\$",
                sol("Answer", "x^{$ans}")
            ),
            time=75
        )
        
    elseif problem_type == 5
        # Negative exponent: x^(-a) = 1/x^a
        a = randint(2, 8)
        
        problem(
            question="Simplify using positive exponents: \$x^{-$a}\$",
            answer="1/x^$a",
            difficulty=(800, 1000),
            solution=steps(
                "Use negative exponent rule: \$x^{-a} = \\frac{1}{x^a}\$",
                sol("Rewrite with positive exponent", "\\frac{1}{x^{$a}}"),
                sol("Answer", "\\frac{1}{x^{$a}}")
            ),
            time=60
        )
        
    else
        # Zero exponent: x^0 = 1
        coefficient = nonzero(-9, 9)
        
        problem(
            question="Simplify: \$($coefficient x)^0\$",
            answer=1,
            difficulty=(700, 850),
            solution=steps(
                "Use the zero exponent rule: \$a^0 = 1\$ for any \$a \\neq 0\$",
                "Any non-zero expression raised to the power 0 equals 1",
                sol("Answer", 1)
            ),
            time=30
        )
    end
end